src/components/input/index.tsx

```diff
@@ -18,6 +18,7 @@ export class SmoothlyInput implements Clearable, Input, Editable {
 	private keepFocusOnReRender = false
 	private lastValue: any
 	private state: Readonly<tidily.State> & Readonly<tidily.Settings>
+	readOnlyAtLoad = false
```

Comment:
other "normal" class variables are marked as private. maybe mark this one as private as well.

Reply:
also move this one down the list and use `private readonlyAtLoad = this.readonly` to avoid the code in componentWillLoad. (this must be bellow the declaration of readonly

Reply:
fixed!